src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.ps1

```diff
@@ -138,6 +138,26 @@ Rule 'Azure.AKS.AzureRBAC' -Type 'Microsoft.ContainerService/managedClusters' -T
     $Assert.HasFieldValue($TargetObject, 'Properties.aadProfile.enableAzureRbac', $True);
 }
 
+# Synopsis: Use Autoscaling to ensure AKS cluster is running efficiently with the right number of nodes for the workloads present.
+Rule 'Azure.AKS.AutoScaling' -Type 'Microsoft.ContainerService/managedClusters', 'Microsoft.ContainerService/managedClusters/agentPools' -Tag @{ release = 'GA'; ruleSet = '2021_09'; } {
+    $agentPools = @(GetAgentPoolProfiles);
+
+    if ($agentPools.Length -eq 0) {
+        return $Assert.Pass();
+    }
+
+    foreach ($agentPool in $agentPools) {
+
+        # Autoscaling only available on virtual machine scale sets
+        if ($Assert.HasFieldValue($agentPool, 'type', 'VirtualMachineScaleSets')) {
```

Comment:
I think this line is not going to be performing how you intended.

`$Assert.HasFieldValue` is going to be returning an `AssertResult` which is always an object, and will also pass the null check on this line.
To get the result of `HasFieldValue`, use `$Assert.HasFieldValue($agentPool, 'type', 'VirtualMachineScaleSets').Result` or explictilty cast to a boolean. i.e. `[bool]$Assert.HasFieldValue($agentPool, 'type', 'VirtualMachineScaleSets')`. Personally I prefer the `.Result` option because it is clearer.

This is why `cluster-A`, and `cluster-B` in the unit tests are reporting as fail instead of pass.

The approach of only reporting on scale set node pools make sense to me.

Reply:
@BernieWhite Ah that makes sense, I had a look at the `AssertRule` class and didn't see that Result property: https://github.com/microsoft/PSRule/blob/main/src/PSRule/Runtime/AssertResult.cs#L33-L36

Reply:
Should be fixed now